# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/nsExternalHelperAppService.cpp
# Commit: 9dc679970296
# Full Hash: 9dc679970296c93353800f9765d023b24fbfc423
# Author: Sarah Ukoha <ukohasarah@gmail.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1555637
# File Overlap Count: 1
# Description:
#   Bug 1555637 - Ask the user what to do for local files, if the file is set to save to disk automatically r=Gijs
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114045
# ==============================================================================

diff -r e72cd321bbaf -r 9dc679970296 uriloader/exthandler/nsExternalHelperAppService.cpp
--- a/uriloader/exthandler/nsExternalHelperAppService.cpp	Tue May 18 09:06:13 2021 +0000
+++ b/uriloader/exthandler/nsExternalHelperAppService.cpp	Tue May 18 09:16:28 2021 +0000
@@ -1862,7 +1862,13 @@
     alwaysAsk = false;
     action = nsIMIMEInfo::saveToDisk;
   }
-
+  // Additionally, if we are asked by the OS to open a local file,
+  // automatically downloading it to create a second copy of that file doesn't
+  // really make sense. We should ask the user what they want to do.
+  if (mSourceUrl->SchemeIs("file") && !alwaysAsk &&
+      action == nsIMIMEInfo::saveToDisk) {
+    alwaysAsk = true;
+  }
   if (alwaysAsk) {
     // Display the dialog
     mDialog = do_CreateInstance(NS_HELPERAPPLAUNCHERDLG_CONTRACTID, &rv);